modules/private/class-jetpack-private.php

```diff
@@ -0,0 +1,349 @@
+			return;
+		}
+
+		include JETPACK__PLUGIN_DIR . '/modules/private/private.php';
+
+		exit;
+	}
+
+	/**
+	 * Does not check whether the blog is private. Accepts blog and user in various types.
+	 * Returns true for super admins.
+	 *
+	 * @param int $blog Current WordPress blod id..
+	 */
+	private static function is_private_blog_user( $blog ) {
+		if ( is_numeric( $blog ) ) {
+			$blog_id = intval( $blog );
+		} elseif ( is_object( $blog ) ) {
+			$blog_id = $blog->blog_id;
+		} elseif ( is_string( $blog ) ) {
+			$fields  = array(
+				'domain' => $blog,
+				'path'   => '/',
+			);
+			$blog    = get_blog_details( $fields );
+			$blog_id = $blog->blog_id;
+		} else {
+			$blog_id = get_current_blog_id();
+		}
+
+		/**
+		 * Filter the capabilites a user needs to have to see the site
+		 *
+		 * @module private
+		 *
+		 * @since 7.4.0
+		 *
+		 * @param string $cap The lowest capability a user needs to have
+		 */
+		$capability = apply_filters( 'jetpack_private_capability', 'read' );
+		return current_user_can_for_blog( $blog_id, $capability );
+	}
+
+	/**
+	 * Hides the blog's name on the login form for private blogs.
+	 */
+	public static function privatize_blog_maybe_mask_blog_name() {
+		add_filter( 'bloginfo', array( __CLASS__, 'privatize_blog_mask_blog_name' ), 3, 2 );
+	}
+
+	/**
+	 * Replaces the the blog's "name" value with "Private Site"
+	 *
+	 * @see privatize_blog_maybe_mask_blog_name()
+	 * @param mixed $value The requested non-URL site information.
+	 * @param mixed $what  Type of information requested.
+	 */
+	public static function privatize_blog_mask_blog_name( $value, $what ) {
+		if ( in_array( $what, array( 'name', 'title' ), true ) ) {
+			$value = __( 'Private Site', 'jetpack' );
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Remove the privatize_blog_mask_blog_name filter
+	 */
+	public static function remove_privatize_blog_mask_blog_name_filter() {
+		remove_filter( 'bloginfo', array( __CLASS__, 'privatize_blog_mask_blog_name' ) );
+	}
+
+	/**
+	 * Filters new comments so that users can't comment on private blogs
+	 *
+	 * @param array $comment Documented in wp-includes/comment.php.
+	 */
+	public static function privatize_blog_comments( $comment ) {
+		self::privatize_blog( null );
+		return $comment;
+	}
+
+	/**
+	 * Extend the 'Site Visibility' privacy options to also include a private option
+	 **/
+	public static function privatize_blog_priv_selector() {
+		?>
+		<style>
+			.jetpack-private__setting-disabled {
+				font-weight: bold;	
+				padding: 10px;
+			}
+			.option-site-visibility fieldset {
+				display: none;
+			}
+		</style>
+		</fieldset>
+		<div class="jetpack-private__setting-disabled highlight">
+			<?php
+			wp_kses(
+				printf(
+					/* translators: URL for Jetpack dashboard. */
+					__( 'This setting is ignored because you <a href="%s">made your site private</a>', 'jetpack' ), // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

Comment:
Should this have a `.`?
```suggestion
					__( 'This setting is ignored because you <a href="%s">made your site private</a>.', 'jetpack' ), // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```

Reply:
Fixed in [11bd70d](https://github.com/Automattic/jetpack/pull/10333/commits/11bd70d0ce5980dc54f310329ee4db8946517ad8).